rest/src/main/java/at/jku/isse/ecco/rest/RepositoryHandler.java

```diff
@@ -0,0 +1,158 @@
+        }
+        return new RestRepository(service, rId, name);
+    }
+
+    public void createRepository() {
+        service = new EccoService();
+        service.setRepositoryDir(path.resolve(".ecco"));
+        service.setBaseDir(path);
+        service.init();
+        new RestRepository(service, rId, name);
+    }
+
+    // Commit ----------------------------------------------------------------------------------------------------------
+    public void addCommit (String message, String config, Path commitFolder, String committer) {
+        service.setBaseDir(commitFolder);
+        service.commit(message, config, committer);
+    }
+
+    //checkout
+    public void checkout(String variantId, Path checkoutPath) {
+        getRepository();
+        service.setBaseDir(checkoutPath);
+        service.checkout(service.getRepository().getVariant(variantId).getConfiguration());
+    }
+
+    // Variant ---------------------------------------------------------------------------------------------------------
+    public RestRepository addVariant(String name, String config, String description){
+        service.addVariant(config, name, description);
+        return getRepository();
+    }
+
+    public RestRepository removeVariant(String variantId){
+        service.removeVariant(variantId);
+        return getRepository();
+    }
+
+    public RestRepository variantSetNameDescription(String variantId, String name, String description) {
+        Variant variant = service.getRepository().getVariant(variantId);
+        variant.setName(name);
+        variant.setDescription(description);
+        service.store();
+        return getRepository();
+    }
+
+    public RestRepository variantAddFeature(String variantId, String featureId){
+
+        List<FeatureRevision> list = new LinkedList<>(Arrays.stream(service.getRepository().getVariant(variantId).getConfiguration().getFeatureRevisions()).toList());
+
+        for(Feature f : service.getRepository().getFeature()){
+            if (f.getId().equals(featureId)) {
+                list.add(f.getLatestRevision());
+            }
+        }
+
+        service.getRepository().getVariant(variantId).getConfiguration().setFeatureRevisions(list.toArray(new FeatureRevision[0]));
+        service.store();
+        return getRepository();
+    }
+
+    public RestRepository variantUpdateFeature(String variantId, String featureName, String id){
+        System.out.println("Update FeatureRevision " + featureName + " from variant " + variantId + " to Revision " + id);
+
+        FeatureRevision[] featureRevisions =  service.getRepository().getVariant(variantId).getConfiguration().getFeatureRevisions();
+        for (int i = 0; i < featureRevisions.length ; i++) {
+            if (featureRevisions[i].getFeature().getName().equals(featureName)){
+                Feature f =  service.getRepository().getFeature().stream().filter(fe -> fe.getName().equals(featureName)).findAny().orElse(null);
```

Comment:
```suggestion
                Feature feature =  service
                  .getRepository()
                  .getFeature()
                  .stream()
                  .filter(feature -> feature.getName().equals(featureName))
                  .findAny()
                  .orElse(null);
```
I am not sure but I think `findAny()` returns null if it cannot find anything anyways. `orElse(null)` might be unnecessary